### PR TITLE
fix(filterTask): made the ui fixes in tasks page filter

### PR DESCRIPTION
**Pull request by simran142002**
### Functionality:
fixed the filter task in tasks page

### Solution:
made changes in filterTasks component

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
ng serve -o
firebase emulators:start

### CODE REVIEW

**Review by Pendyala-thanvi — APPROVED** [review]

File: src/app/body/filter-task/filter-task.component.ts [hunk 1]
```diff
@@ -40,12 +47,63 @@ export class FilterTaskComponent implements OnInit {
...
 
   ngOnInit(): void {
-    this.project = this.defaultProject
-    this.onProjectChange();
+    if(this.startService.showTeamsData) {
+      this.project = this.authService.getTeamId();
+      this.teamIds = this.backendService.getOrganizationTeamIds();
+      this.readTeamData(this.project);
+      this.project = this.defaultProject
+      this.onProjectChange();
```
Comment (sanjaykrishna1203) [inline src/app/body/filter-task/filter-task.component.ts:58]: What is the Use of Calling this Fuction again when we are calling readTeanData()

File: src/app/body/filter-task/filter-task.component.ts [hunk 2]
```diff
@@ -40,12 +47,63 @@ export class FilterTaskComponent implements OnInit {
...
+        if(data){
+          this.startService.applicationDataStateObservable.subscribe((data) => {
+            if(data) {
+              this.appSettings.teamData.subscribe((data) => {
+                if(data) {
+                  this.project = this.authService.getTeamId();
+                  this.teamIds = this.backendService.getOrganizationTeamIds();
+                  this.readTeamData(this.project);
+                  this.project = this.defaultProject
+                  this.onProjectChange();
```
Comment (sanjaykrishna1203) [inline src/app/body/filter-task/filter-task.component.ts:70]: Here too

**Review by sanjaykrishna1203 — CHANGES_REQUESTED** [review]

**Review by sanjaykrishna1203 — APPROVED** [review]
Approved

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**